core/src/main/scala/kafka/raft/KafkaNetworkChannel.scala

```diff
@@ -105,14 +108,18 @@ class KafkaNetworkChannel(
   private val correlationIdCounter = new AtomicInteger(0)
   private val endpoints = mutable.HashMap.empty[Int, Node]
 
-  private val requestThread = new RaftSendThread(
+  private var requestThread = new RaftSendThread(
     name = threadNamePrefix + "-outbound-request-thread",
     networkClient = client,
     requestTimeoutMs = requestTimeoutMs,
     time = time,
     isInterruptible = false
   )
 
+  def setRequestThread(raftSendThread: RaftSendThread): Unit = {
+    requestThread = raftSendThread
+  }
```

Comment:
Hum.. I would rather prefer to avoid doing this. In the test, it seems that you could rely on the mocked client instead of doing this. Could you please give it a try?

Reply:
Yes, you are right. Updated.